modules/phase_field/src/kernels/CHSplitConcentration.C

```diff
@@ -0,0 +1,66 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "CHSplitConcentration.h"
+
+template<>
+InputParameters validParams<CHSplitConcentration>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Concentration kernel in Split Cahn-Hilliard that solves chemical potential in a weak form");
+  params.addRequiredParam<MaterialPropertyName>("mobility", "Mobility property name");
+  params.addRequiredCoupledVar("chemical_potential_var", "Chemical potential variable");
+
+  return params;
+}
+
+CHSplitConcentration::CHSplitConcentration(const InputParameters & parameters) :
+    DerivativeMaterialInterface< Kernel >(parameters),
+    _mobility_name(getParam<MaterialPropertyName>("mobility")),
+    _mobility(getMaterialProperty<RealTensorValue>(_mobility_name)),
+    _dmobility_dc(getMaterialPropertyDerivative<RealTensorValue>(_mobility_name, name())),
+    _mu_var(coupled("chemical_potential_var")),
+    _grad_mu(coupledGradient("chemical_potential_var"))
+{
+}
+
+Real
+CHSplitConcentration::computeQpResidual()
+{
+  RealVectorValue a;
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+    for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+      a(i) += _mobility[_qp](i, j) * _grad_mu[_qp](j);
```

Comment:
`TypeTensor * TypeVector` already exists.  You should be able to replace these hand-coded loops with:
```
RealVectorValue a = _mobility[_qp] * _grad_mu[_qp]
```